improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+            else:
+                codes = attrs[self.model_id_attr].split(" ")
+                names = [MODEL_NAMES[code] for code in codes]
+                self.model = ", ".join(names)
+
+        else:
+            if self.model_id_attr in attrs:
+                for key in MODEL_CODES:
+                    if (
+                        f"{key} Model" in attrs["title"]
+                        and attrs[self.model_id_attr] != MODEL_CODES[key]
+                    ):
+                        self.errors.append(
+                            f"Title {attrs['title']} is inconsistent with model ID "
+                            f"attribute {attrs[self.model_id_attr]}"
+                        )
+
+                try:
+                    self.model = MODEL_NAMES[attrs[self.model_id_attr]]
+                except KeyError:
+                    self.errors.append(
+                        f"Attribute {attrs[self.model_id_attr]} is not a valid single "
+                        "model.  If valid for blend, then title attribute is missing "
+                        f"expected substring {BLEND_TITLE_SUBSTR}."
+                    )
+
+    def check_attributes(self, attrs):
+        """Checks for unexpected attributes, then interprets values for model
+        information and checks for self-consistency"""
+        try:
+            permitted_attributes = COMP_ATTRS + DIAG_ATTRS[self.diagnostic]
+        except KeyError:
+            permitted_attributes = COMP_ATTRS.copy()
+
+        if any([attr in NONCOMP_ATTRS for attr in attrs]):
+            self.errors.append(
+                f"Attributes {attrs.keys()} include one or more forbidden "
+                f"values {NONCOMP_ATTRS}"
+            )
+        elif any([attr not in permitted_attributes for attr in attrs]):
+            self.warnings.append(
+                f"{attrs.keys()} include unexpected attributes. Please check the "
+                "standard to ensure this is valid."
+            )
+
+        if self.diagnostic in DIAG_ATTRS:
+            required = DIAG_ATTRS[self.diagnostic]
+            if any([req not in attrs for req in required]):
+                self.errors.append(
+                    f"Attributes {attrs.keys()} missing one or more required "
+                    f"values {required}"
+                )
+
+        if self.field_type != self.ANCIL:
+            if not all([attr in attrs for attr in MANDATORY_ATTRIBUTES]):
+                self.errors.append(
+                    f"Attributes {attrs.keys()} missing one or more mandatory "
+                    f"values {MANDATORY_ATTRIBUTES}"
+                )
+
+            try:
+                self.post_processed = (
+                    True
+                    if PP_TITLE_SUBSTR in attrs["title"]
+                    or BLEND_TITLE_SUBSTR in attrs["title"]
+                    else False
+                )
+            except KeyError:
+                self.errors.append("Cube is missing mandatory title attribute")
```

Comment:
This is unnecessary because this method has just checked that the title attribute is present. It would be better to change the "try" to be `if "title" in attrs:` and delete these two lines (and the else block becomes part of the if block).

Reply:
OK